app/javascript/components/workflows/professor/exams/new/editor/components/questions/CodeTag.tsx

```diff
@@ -142,33 +143,77 @@ const FileModal: React.FC<FileModalProps> = (props) => {
   );
 };
 
-interface CodeTagProps {
-  qnum: number;
-  pnum: number;
-  bnum: number;
-  info: CodeTagInfo;
-  value: CodeTagState;
-  onChange: (newInfo: CodeTagInfo, newVal: CodeTagState) => void;
-  disabled: boolean;
-}
-
-const CodeTag: React.FC<CodeTagProps> = (props) => {
+const EditChoices: React.FC<WrappedFieldsProps> = (props) => {
+  const {
+    choices,
+    answer,
+  } = props;
+  const { input } = choices;
   const {
-    qnum,
-    pnum,
-    bnum,
-    info,
     value,
     onChange,
-    disabled,
+  }: {
+    value: CodeTagInfo['choices'];
+    onChange: (newVal: CodeTagInfo['choices']) => void;
+  } = input;
+  return (
+    <>
+      <Form.Label column sm={2}>Files source</Form.Label>
+      <Col sm={10}>
+        <ButtonGroup>
+          <Button
+            variant={value === 'exam' ? 'secondary' : 'outline-secondary'}
+            active={value === 'exam'}
+            onClick={(): void => {
+              onChange('exam');
+              answer.input.onChange({ NO_ANS: true });
+            }}
+          >
+            Files for full exam
+          </Button>
+          <Button
+            variant={value === 'question' ? 'secondary' : 'outline-secondary'}
+            active={value === 'question'}
+            onClick={(): void => {
+              onChange('question');
+              answer.input.onChange({ NO_ANS: true });
+            }}
+          >
+            Files for current question
+          </Button>
+          <Button
+            variant={value === 'part' ? 'secondary' : 'outline-secondary'}
+            active={value === 'part'}
+            onClick={(): void => {
+              onChange('part');
+              answer.input.onChange({ NO_ANS: true });
+            }}
+          >
+            Files for current part
+          </Button>
+        </ButtonGroup>
+      </Col>
+    </>
+  );
+};
+
+const EditAnswer: React.FC<WrappedFieldsProps> = (props) => {
+  const {
+    answer,
+    choices,
   } = props;
-  const { choices, prompt } = info;
+  const { input } = answer;
+  const {
+    value,
+    onChange,
+  } = input;
   const [showModal, setShowModal] = useState(false);
   const examReferences = useContext(ExamFilesContext);
   const questionReferences = useContext(QuestionFilesContext);
   const partReferences = useContext(PartFilesContext);
   let references: FileRef[];
-  switch (choices) {
+  const choice: CodeTagInfo['choices'] = choices.input.value;
```

Comment:
Again a weird singular/plural inconsistency.  We should rename and be consistent...